### PR TITLE
Skip `_extends` helper for `{…x}` with modern targets

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

The `_extends` helper already delegates to `Object.assign` when it is available.
If we statically know that `Object.assign` exists (using the user's `targets`), we can avoid injecting `_extends`.

### CODE REVIEW

File: packages/babel-plugin-proposal-object-rest-spread/test/fixtures/assumption-setSpreadProperties/targets-support-object-assign/output.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+let a = Object.assign({
+  x: 1
+}, b);
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-proposal-object-rest-spread/test/fixtures/assumption-setSpreadProperties/targets-support-object-assign/output.js:3]: Without this PR, the output would be something like
```js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

let a = _extends({
  x: 1
}, b);
```

**Review by JLHwung — APPROVED** [review]
The future is on the horizon: `preset-env` will be simplified to a static array of babel plugins.

**Comment by nicolo-ribaudo** [comment]
Yes! :smile:

**Review by existentialism — APPROVED** [review]
💯

**Pull request merged**